executor/runtime/docker/docker_linux.go

```diff
@@ -208,8 +134,14 @@ func setupSystemServices(parentCtx context.Context, c runtimeTypes.Container, cf
 	}
 	defer conn.Close()
 
-	for _, svc := range systemServices {
-		if svc.enabledCheck != nil && !svc.enabledCheck(&cfg, c) {
+	// TODO: it would be nice not to fetch this twice
+	sidecars, err := c.SidecarConfigs()
+	if err != nil {
+		return err
+	}
+	// TODO: Can we somehow make sure titus-container always starts first?
+	for _, svc := range sidecars {
+		if svc.EnabledCheck != nil && !svc.EnabledCheck(&cfg, c) {
```

Comment:
Can `EnabledCheck` be nil? If so, you'll also have to test when you create volume containers, for example.

Reply:
I think in practice we happen to not have any sidecars that have volumes but *not* a enabledCheck.
But good idea, I'll make sure to test just in case we ever have one.